plugins/terraform/layer0/data_source_layer0_load_balancer_test.go

```diff
@@ -0,0 +1,59 @@
+package layer0
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/client"
+	"github.com/quintilesims/layer0/client/mock_client"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDataSourceLayer0LoadBalancerRead(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockClient := mock_client.NewMockClient(ctrl)
+
+	query := url.Values{}
+	query.Set(client.TagQueryParamType, "load_balancer")
+	query.Set(client.TagQueryParamName, "lb_name")
+	query.Set(client.TagQueryParamEnvironmentID, "env_id")
+
+	mockClient.EXPECT().
+		ListTags(query).
+		Return([]models.Tag{{EntityID: "lb_id"}}, nil)
+
+	loadBalancer := &models.LoadBalancer{
+		LoadBalancerID:   "lb_id",
+		LoadBalancerName: "lb_name",
+		EnvironmentID:    "env_id",
+		EnvironmentName:  "env_name",
+		IsPublic:         false,
+		URL:              "some_url",
+	}
+
+	mockClient.EXPECT().
+		ReadLoadBalancer("lb_id").
+		Return(loadBalancer, nil)
+
+	loadBalancerDataSource := Provider().(*schema.Provider).DataSourcesMap["layer0_load_balancer"]
+	d := schema.TestResourceDataRaw(t, loadBalancerDataSource.Schema, map[string]interface{}{
+		"name":           "lb_name",
+		"environment_id": "env_id",
+	})
+
+	if err := dataSourceLayer0LoadBalancerRead(d, mockClient); err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, "lb_id", d.Id())
+	assert.Equal(t, "lb_name", d.Get("name"))
+	assert.Equal(t, "env_id", d.Get("environment_id"))
+	assert.Equal(t, "env_name", d.Get("environment_name"))
+	assert.Equal(t, true, d.Get("private"))
```

Comment:
[nit] odd that our model used `isPublic` but we expose `private` via terraform.

Reply:
This is a UX decision, the same reason we expose `--private` in the CLI; we want the "default" behavior (omission of a flag/field) to be public.